overlord/snapstate/backend/copydata.go

```diff
@@ -108,6 +109,40 @@ func (b Backend) UndoCopySnapData(newInfo, oldInfo *snap.Info, _ progress.Meter,
 	return firstErr(err1, err2)
 }
 
+func (b Backend) SetupSnapSaveData(info *snap.Info, meter progress.Meter) error {
+	// ubuntu-save is only present on core images
+	if release.OnClassic {
+		return nil
+	}
+
+	// verify that ubuntu-save has been mounted under the expected path and
+	// that it is indeed a mount-point.
+	if hasSave, err := osutil.IsMounted(dirs.SnapSaveDir); err != nil || !hasSave {
+		if err != nil {
+			return fmt.Errorf("cannot check if ubuntu-save is mounted: %v", err)
+		}
+		return nil
+	}
+
+	saveDir := snap.CommonDataSaveDir(info.InstanceName())
+	return os.MkdirAll(saveDir, 0755)
```

Comment:
I think that this makes sense, but I'm writing just to make sure it's intended (maybe a comment could be added to make this explicit): if the permission is 0755, it means that only the root use can create files under ubuntu-save.

Reply:
I probably just need @pedronis to verify this should be like this, my understanding is that this is the correct behaviour for writing files to the folder on ubuntu-save, but I would love a second thumbs up on this

Reply:
should behave like /var/snap/common which I think is also like this